cli/src/semdep/parsers/gem.py

```diff
@@ -70,7 +71,7 @@
 def parse_gemfile(
     lockfile_path: Path, manifest_path: Optional[Path]
 ) -> List[FoundDependency]:
-    deps_opt = safe_path_parse(lockfile_path, gemfile)
+    deps_opt = safe_path_parse(lockfile_path, gemfile, ParserName("gemfile.lock"))
```

Comment:
For each call to safe_path_parse, added an identifier so that we know what parser we tried to use on the given path